modules/phase_field/src/userobjects/PolycrystalVoronoi.C

```diff
@@ -55,26 +57,37 @@ void
 PolycrystalVoronoi::getGrainsBasedOnPoint(const Point & point,
                                           std::vector<unsigned int> & grains) const
 {
+  grains.resize(0);
+  Real d_min = _range.norm();
+  Real distance;
   auto n_grains = _centerpoints.size();
-  auto min_distance = _range.norm();
   auto min_index = n_grains;
 
-  // Loops through all of the grain centers and finds the center that is closest to the point p
+  // Find the closest centerpoint to the current point
   for (MooseIndex(_centerpoints) grain = 0; grain < n_grains; ++grain)
   {
-    auto distance = _mesh.minPeriodicDistance(_vars[0]->number(), _centerpoints[grain], point);
-
-    if (distance < min_distance)
+    distance = _mesh.minPeriodicDistance(_vars[0]->number(), _centerpoints[grain], point);
+    if (distance < d_min)
     {
-      min_distance = distance;
+      d_min = distance;
       min_index = grain;
     }
   }
-
   mooseAssert(min_index < n_grains, "Couldn't find closest Voronoi cell");
-
-  grains.resize(1);
-  grains[0] = min_index;
+  Point current_grain = _centerpoints[min_index];
+  grains.push_back(min_index); //closest centerpoint always gets included
+
+  if (_int_width > 0.0)
+    for (MooseIndex(_centerpoints) grain = 0; grain < n_grains; ++grain)
+      if (grain != min_index)
+      {
+        Point next_grain = _centerpoints[grain];
+        Point N = findNormalVector(point, current_grain, next_grain);
+        Point cntr = findCenterPoint(point, current_grain, next_grain);
+        distance = N * (cntr - point);
+        if (distance < 0.6 * _int_width)
```

Comment:
what is the significance of `0.6` here?

Reply:
I've been playing with limiting the range to lower the computational cost, but I think the profile works better without this so I am removing it.